.golangci.yml

```diff
@@ -91,4 +91,7 @@ issues:
   exclude-rules:
     - linters:
         - typecheck
-      text: "could not import C"
\ No newline at end of file
+      text: "could not import C"
+    - linters:
```

Comment:
Do you really need to disable the linter here?

Reply:
it just skip the lint for the specific file. the test cases are very similar, hence I don't know the reason why dupl mark as duplicated.

Reply:
I know that it might be challenging, but I would prefer to do not disable the linter.
If there are duplicated tests, we should probably have only one test.

Reply:
```
pkg/collector/metric/types/types_test.go:97: 97-109 lines are duplicate of `pkg/collector/metric/types/types_test.go:110-122` (dupl)
                It("SetAggrStat", func() {
                        instance := types.UInt64StatCollection{
                                Stat: make(map[string]*types.UInt64Stat),
                        }
                        instance.SetAggrStat("SetAggrStat", uint64(1))
                        Expect(instance.Stat["SetAggrStat"].Aggr).To(Equal(uint64(1)))
                        instance.SetAggrStat("SetAggrStat", uint64(2))
                        Expect(instance.Stat["SetAggrStat"].Aggr).To(Equal(uint64(2)))
                        Expect(instance.Stat["SetAggrStat"].Delta).To(Equal(uint64(1)))
                        instance.SetAggrStat("SetAggrStat", uint64(0))
                        Expect(instance.Stat["SetAggrStat"].Aggr).To(Equal(uint64(2)))
                        Expect(instance.Stat["SetAggrStat"].Delta).To(Equal(uint64(1)))
                })
pkg/collector/metric/types/types_test.go:110: 110-122 lines are duplicate of `pkg/collector/metric/types/types_test.go:123-135` (dupl)
                It("AddDeltaStat", func() {
                        instance := types.UInt64StatCollection{
                                Stat: make(map[string]*types.UInt64Stat),
                        }
                        instance.AddDeltaStat("AddDeltaStat", uint64(1))
                        Expect(instance.Stat["AddDeltaStat"].Aggr).To(Equal(uint64(1)))
                        instance.AddDeltaStat("AddDeltaStat", uint64(2))
                        Expect(instance.Stat["AddDeltaStat"].Aggr).To(Equal(uint64(3)))
                        Expect(instance.Stat["AddDeltaStat"].Delta).To(Equal(uint64(3)))
                        instance.AddDeltaStat("AddDeltaStat", uint64(0))
                        Expect(instance.Stat["AddDeltaStat"].Aggr).To(Equal(uint64(3)))
                        Expect(instance.Stat["AddDeltaStat"].Delta).To(Equal(uint64(3)))
                })
pkg/collector/metric/types/types_test.go:123: 123-135 lines are duplicate of `pkg/collector/metric/types/types_test.go:97-109` (dupl)
                It("SetDeltaStat", func() {
                        instance := types.UInt64StatCollection{
                                Stat: make(map[string]*types.UInt64Stat),
                        }
                        instance.SetDeltaStat("SetDeltaStat", uint64(1))
                        Expect(instance.Stat["SetDeltaStat"].Aggr).To(Equal(uint64(1)))
                        instance.SetDeltaStat("SetDeltaStat", uint64(2))
                        Expect(instance.Stat["SetDeltaStat"].Aggr).To(Equal(uint64(3)))
                        Expect(instance.Stat["SetDeltaStat"].Delta).To(Equal(uint64(2)))
                        instance.SetDeltaStat("SetDeltaStat", uint64(0))
                        Expect(instance.Stat["SetDeltaStat"].Aggr).To(Equal(uint64(3)))
                        Expect(instance.Stat["SetDeltaStat"].Delta).To(Equal(uint64(2)))
                })
pkg/collector/metric/types/types_test.go:136: 136-151 lines are duplicate of `pkg/collector/metric/types/types_test.go:152-167` (dupl)
                It("SumAllDeltaValues", func() {
                        instance := types.UInt64StatCollection{
                                Stat: make(map[string]*types.UInt64Stat),
                        }
                        value := instance.SumAllDeltaValues()
                        Expect(value).To(Equal(uint64(0)))
                        instance.SetDeltaStat("SumAllDeltaValues", uint64(2))
                        value = instance.SumAllDeltaValues()
                        Expect(value).To(Equal(uint64(2)))
                        instance.SetDeltaStat("SumAllDeltaValues", uint64(0))
                        value = instance.SumAllDeltaValues()
                        Expect(value).To(Equal(uint64(2)))
                        instance.SetDeltaStat("SumAllDeltaValues1", uint64(2))
                        value = instance.SumAllDeltaValues()
                        Expect(value).To(Equal(uint64(4)))
                })
pkg/collector/metric/types/types_test.go:152: 152-167 lines are duplicate of `pkg/collector/metric/types/types_test.go:136-151` (dupl)
                It("SumAllAggrValues", func() {
                        instance := types.UInt64StatCollection{
                                Stat: make(map[string]*types.UInt64Stat),
                        }
                        value := instance.SumAllAggrValues()
                        Expect(value).To(Equal(uint64(0)))
                        instance.SetAggrStat("SumAllAggrValues", uint64(2))
                        value = instance.SumAllAggrValues()
                        Expect(value).To(Equal(uint64(2)))
                        instance.SetAggrStat("SumAllAggrValues", uint64(0))
                        value = instance.SumAllAggrValues()
                        Expect(value).To(Equal(uint64(2)))
                        instance.SetAggrStat("SumAllAggrValues1", uint64(2))
                        value = instance.SumAllAggrValues()
                        Expect(value).To(Equal(uint64(4)))
                })
```
if we enable the lint to this test file. well, in fact each test run 3 times to cover some case.(for example: 1st time invoke, 2nd time value update and 3rd time check with specific value as 0) considering the function name a similar with each other then the dupl detected it and reported. @rootfs , @marceloamaral  any suggestions?

Reply:
The only line that is duplicated seems to be:
```golang
instance := types.UInt64StatCollection{
  Stat: make(map[string]*types.UInt64Stat),
}
```

Did you try to set the variable in the before each function?
```golang
var instance  types.UInt64StatCollection

BeforeEach(func() {
  instance = types.UInt64StatCollection{
  Stat: make(map[string]*types.UInt64Stat),
}
})

It("SetAggrStat", func() { [...]
```

Reply:
```
pkg/collector/metric/types/types_test.go:134: 134-146 lines are duplicate of `pkg/collector/metric/types/types_test.go:147-159` (dupl)
                It("SumAllDeltaValues", func() {
                        value := instance.SumAllDeltaValues()
                        Expect(value).To(Equal(uint64(0)))
                        instance.SetDeltaStat("SumAllDeltaValues", uint64(2))
                        value = instance.SumAllDeltaValues()
                        Expect(value).To(Equal(uint64(2)))
                        instance.SetDeltaStat("SumAllDeltaValues", uint64(0))
                        value = instance.SumAllDeltaValues()
                        Expect(value).To(Equal(uint64(2)))
                        instance.SetDeltaStat("SumAllDeltaValues1", uint64(2))
                        value = instance.SumAllDeltaValues()
                        Expect(value).To(Equal(uint64(4)))
                })
pkg/collector/metric/types/types_test.go:147: 147-159 lines are duplicate of `pkg/collector/metric/types/types_test.go:134-146` (dupl)
                It("SumAllAggrValues", func() {
                        value := instance.SumAllAggrValues()
                        Expect(value).To(Equal(uint64(0)))
                        instance.SetAggrStat("SumAllAggrValues", uint64(2))
                        value = instance.SumAllAggrValues()
                        Expect(value).To(Equal(uint64(2)))
                        instance.SetAggrStat("SumAllAggrValues", uint64(0))
                        value = instance.SumAllAggrValues()
                        Expect(value).To(Equal(uint64(2)))
                        instance.SetAggrStat("SumAllAggrValues1", uint64(2))
                        value = instance.SumAllAggrValues()
                        Expect(value).To(Equal(uint64(4)))
                })
pkg/collector/metric/types/types_test.go:104: 104-113 lines are duplicate of `pkg/collector/metric/types/types_test.go:114-123` (dupl)
                It("SetAggrStat", func() {
                        instance.SetAggrStat("SetAggrStat", uint64(1))
                        Expect(instance.Stat["SetAggrStat"].Aggr).To(Equal(uint64(1)))
                        instance.SetAggrStat("SetAggrStat", uint64(2))
                        Expect(instance.Stat["SetAggrStat"].Aggr).To(Equal(uint64(2)))
                        Expect(instance.Stat["SetAggrStat"].Delta).To(Equal(uint64(1)))
                        instance.SetAggrStat("SetAggrStat", uint64(0))
                        Expect(instance.Stat["SetAggrStat"].Aggr).To(Equal(uint64(2)))
                        Expect(instance.Stat["SetAggrStat"].Delta).To(Equal(uint64(1)))
                })
pkg/collector/metric/types/types_test.go:114: 114-123 lines are duplicate of `pkg/collector/metric/types/types_test.go:124-133` (dupl)
                It("AddDeltaStat", func() {
                        instance.AddDeltaStat("AddDeltaStat", uint64(1))
                        Expect(instance.Stat["AddDeltaStat"].Aggr).To(Equal(uint64(1)))
                        instance.AddDeltaStat("AddDeltaStat", uint64(2))
                        Expect(instance.Stat["AddDeltaStat"].Aggr).To(Equal(uint64(3)))
                        Expect(instance.Stat["AddDeltaStat"].Delta).To(Equal(uint64(3)))
                        instance.AddDeltaStat("AddDeltaStat", uint64(0))
                        Expect(instance.Stat["AddDeltaStat"].Aggr).To(Equal(uint64(3)))
                        Expect(instance.Stat["AddDeltaStat"].Delta).To(Equal(uint64(3)))
                })
pkg/collector/metric/types/types_test.go:124: 124-133 lines are duplicate of `pkg/collector/metric/types/types_test.go:104-113` (dupl)
                It("SetDeltaStat", func() {
                        instance.SetDeltaStat("SetDeltaStat", uint64(1))
                        Expect(instance.Stat["SetDeltaStat"].Aggr).To(Equal(uint64(1)))
                        instance.SetDeltaStat("SetDeltaStat", uint64(2))
                        Expect(instance.Stat["SetDeltaStat"].Aggr).To(Equal(uint64(3)))
                        Expect(instance.Stat["SetDeltaStat"].Delta).To(Equal(uint64(2)))
                        instance.SetDeltaStat("SetDeltaStat", uint64(0))
                        Expect(instance.Stat["SetDeltaStat"].Aggr).To(Equal(uint64(3)))
                        Expect(instance.Stat["SetDeltaStat"].Delta).To(Equal(uint64(2)))
                })
```
after the change... well I don't understand why dupl mark dupl by function name.

Reply:
I suppose we should mark here as false positive... https://golangci-lint.run/usage/false-positives/ 
or should we open a bug to dupl?

Reply:
mark as false positive as
I don't want to change all the settings. @marceloamaral , for example remove Delta/Aggr related functions. if do so, which should in anther PR.